src/main/java/edu/ie3/datamodel/io/naming/EntityPersistenceNamingStrategy.java

```diff
@@ -125,17 +125,46 @@
   }
 
   /**
-   * Extracts meta information from a valid file name for an individual time series
+   * Extracts meta information from a valid source name for an individual time series
    *
-   * @param fileName File name to extract information from
-   * @return Meta information form individual time series file name
+   * @param sourceName Name of the source to extract information from, e.g. file name or SQL table
+   *     name
+   * @return Meta information form individual time series source name
+   * @deprecated since 3.0. Use {@link #getIndividualTimesSeriesMetaInformation(String)} instead
    */
-  public IndividualTimeSeriesMetaInformation extractIndividualTimesSeriesMetaInformation(
-      String fileName) {
-    Matcher matcher = getIndividualTimeSeriesPattern().matcher(fileName);
+  @Deprecated(since = "3.0", forRemoval = true)
+  public edu.ie3.datamodel.io.csv.timeseries.IndividualTimeSeriesMetaInformation
+      extractIndividualTimesSeriesMetaInformation(String sourceName) {
+    Matcher matcher = getIndividualTimeSeriesPattern().matcher(sourceName);
+    if (!matcher.matches())
+      throw new IllegalArgumentException(
+          "Cannot extract meta information on individual time series from '" + sourceName + "'.");
+
+    String columnSchemeKey = matcher.group("columnScheme");
+    edu.ie3.datamodel.io.csv.timeseries.ColumnScheme columnScheme =
+        edu.ie3.datamodel.io.csv.timeseries.ColumnScheme.parse(columnSchemeKey)
+            .orElseThrow(
+                () ->
+                    new IllegalArgumentException(
+                        "Cannot parse '" + columnSchemeKey + "' to valid column scheme."));
+
+    return new edu.ie3.datamodel.io.csv.timeseries.IndividualTimeSeriesMetaInformation(
+        UUID.fromString(matcher.group("uuid")), columnScheme);
+  }
+
+  /**
+   * Extracts meta information from a valid source name for an individual time series
+   *
+   * @param sourceName Name of the source to extract information from, e.g. file name or SQL table
+   *     name
+   * @return Meta information form individual time series source name
+   */
+  public IndividualTimeSeriesMetaInformation getIndividualTimesSeriesMetaInformation(
```

Comment:
Within the majority of the refactorings, you full omit the `get` keyword. You might consider doing it here as well!?